src/services/repo/useActivateFlagMeasurements.tsx

```diff
@@ -0,0 +1,111 @@
+import { useMutation, useQueryClient } from '@tanstack/react-query'
+import z from 'zod'
+
+import { renderToast } from 'services/toast'
+import Api from 'shared/api'
+import { NetworkErrorObject } from 'shared/api/helpers'
+
+export const MEASUREMENT_TYPE = {
+  FLAG_COVERAGE: 'FLAG_COVERAGE',
+  COMPONENT_COVERAGE: 'COMPONENT_COVERAGE',
+} as const
+
+interface ActivateFlagMeasurementsProps {
+  provider: string
+  owner: string
+  repo: string
+  measurementType: (typeof MEASUREMENT_TYPE)[keyof typeof MEASUREMENT_TYPE]
+}
+
+const ResponseSchema = z.object({
+  activateMeasurements: z
+    .object({
+      error: z.union([
+        z.object({
+          __typename: z.literal('UnauthenticatedError'),
+          message: z.string(),
+        }),
+        z.object({
+          __typename: z.literal('ValidationError'),
+          message: z.string(),
+        }),
+      ]),
+    })
+    .nullable(),
+})
+
+export function useActivateFlagMeasurements({
+  provider,
+  owner,
+  repo,
+  measurementType,
+}: ActivateFlagMeasurementsProps) {
+  const queryClient = useQueryClient()
+  return useMutation({
+    mutationFn: () => {
+      const query = `
```

Comment:
Nit: maybe we can pull the query const out of the hook so it's only declared the one time rather than every iteration of the hook?

Reply:
why not

Reply:
It's a slight perf optimization, essentially the overhead of defining once and using multiple times vs. defining + cleaning up at each iteration.

Another slight benefit would be readability, since we can condense the stuff within the hook to just be business logic vs. consts + business logic

Reply:
Of course! i really meant "why not let's do it" 😆

Reply:
oop 🙈